modules/dnf/dnf.sh

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env bash
+
+# Tell build process to exit if there are any errors.
+set -euo pipefail
+
+# Fail the build if dnf5 isn't installed
+if ! rpm -q dnf5 &>/dev/null; then
+  echo "ERROR: Main dependency 'dnf5' is not installed. Install 'dnf5' before using this module to solve this error."
+  exit 1
+fi
+
+# Fail the build if dnf5 plugins aren't installed
+if ! rpm -q dnf5-plugins &>/dev/null; then
+  echo "ERROR: Dependency 'dnf5-plugins' is not installed. It is needed for cleanly adding COPR repositories."
+  echo "       Install 'dnf5-plugins' before using this module to solve this error."
+  exit 1
+fi
```

Comment:
Could having this prevent builds from even happening if the user's base image just doesn't have it already installed? They wouldn't be able to use the `dnf` module to install it to fix the issue

Reply:
Hmm, good notice.

I'll just remove this check then